armi/materials/material.py

```diff
@@ -845,9 +851,36 @@ class FuelMaterial(Material):
     class1_wt_frac = None
     class1_custom_isotopics = None
     class2_custom_isotopics = None
-    puFrac = 0.0
-    uFrac = 0.0
-    zrFrac = 0.0
+    puFracDefault = 0.0
+    uFracDefault = 0.0
+    zrFracDefault = 0.0
+
+    def __init__(self):
+        self._puFrac = 0
+        Material._init_()
+
+    @property
+    def puFrac(self):
+        """Get the Pu Frac."""
+        # ideally the parameters would be stored on component, not Block (parent, not grandparent)
+        myBlock = self.getGrandparent()
+        if myBlock is not None:
+            return myBlock.p.puFrac
+        runLog.warning(
+            "{self} has no block attached, deferring to internally stored Pu Frac. "
+            "Storing properties on the material objected will be deprecated in the near future.",
+            single=true,
+        )
+        return self_puFrac
+
+    @puFrac.setter
+    def x(self, puFrac):
+        runLog.warning(
+            "Materials should not store state, so this component will look at its (grand)parent for Pu fraction unless it us detached. "
+            "Storing properties on the material objected will be deprecated in the near future.",
+            single=true,
+        )
+        self._puFrac = puFrac
```

Comment:
When we set the `puFrac` directly, should this assume a reduction in `uFrac` so that the total fraction is 1.0? I wonder if a 1.0 check should be added.